services/core/java/com/android/server/wm/RootWindowContainer.java

```diff
@@ -2488,17 +2493,32 @@ boolean resumeFocusedTasksTopActivities(
     boolean resumeFocusedTasksTopActivities(
             Task targetRootTask, ActivityRecord target, ActivityOptions targetOptions,
             boolean deferPause) {
+        return resumeFocusedTasksTopActivities(targetRootTask, target, targetOptions, deferPause,
+                false /* isAfterUserForegrounded */);
+    }
+
+    boolean resumeFocusedTasksTopActivities(
+            Task targetRootTask, ActivityRecord target, ActivityOptions targetOptions,
+            boolean deferPause, boolean isAfterUserForegrounded) {
         if (!mTaskSupervisor.readyToResume()) {
             return false;
         }
 
         boolean result = false;
-        if (targetRootTask != null && (targetRootTask.isTopRootTaskInDisplayArea()
+        if (!isAfterUserForegrounded && targetRootTask != null &&
+                (targetRootTask.isTopRootTaskInDisplayArea()
```

Comment:
Although this method is currently always called with `targetRootTask == null` when `isAfterUserForegrounded == true`, should this condition also include `shouldResumeForUserForeground(targetRootTask)` similar to the two other checks below?